docs/developers/repository-classes.md

```diff
@@ -0,0 +1,172 @@
+repository instance. This value is set to `default` if not provided. If more than one repository is to be used
+at the same time, it is important to set distinct names.
+
+TUF repository is instantiated lazily the first time it is needed. This object is not meant to be used directly.
+The main purpose of TAF's repository class is to group operations which enable valid update of TUF metadata and acquiring
+information like is a key can be used to sign a certain metadata file or finding roles that are linked with
+the provided public key. To set up a new repository or add a new signing key, it is recommended to use the
+`developer_tool` since it contains full implementations of these complex functionalities. Functionalities
+like updating targets and signing metadata or updating a metadata's expiration date are fully covered by repository
+class's methods and can be used directly. These include:
+- `update_timestamp_keystores`, `update_snapshot_keystores` (`update_rolename_keystores`) and `update_role_keystores` (for delegated roles)
+-`update_timestamp_yubikeys`, `update_snapshot_yubikeys` (`update_rolename_yubikeys`) and `update_role_yubikeys` (for delegated roles)
+
+If `added_targets_data` or `removed_targets_data` is passed in when calling these methods (only applicable to
+`targets` and delegated target roles), information about target files will be updated and the corresponding metadata
+file will be signed. Its expiration date will be updated too. If there is targets data or if the called method
+corresponds to a non-targets role, the metadata file's expiration will still be updated and the file will be signed.
+
+
+## `AuthenticationRepository`
+
+This class is derived from both `GitRepository` and TAF's `Repository`. Authentication repositories are expected
+to contain TUF metadata and target files, but are also Git repositories. It is important to note that only files
+inside the `targets` folder are tracked and secured by TUF.
+
+Instances of the `AuthenticationRepository` are created by passing the same arguments as to `GitRepository` (`library_dir`, `name`, `urls`, `custom`, `default_branch`, `allow_unsafe` and `path` which can replace `library_dir` and `name` combination), as well as some optional additional arguments:
+- `conf_directory_root` - path to the directory where the `last_validated_commit`will be stored. `last_validate_commit` is generated by the updater after a successful update.
+- `out_of_band_authentication` - manually specified initial commit, used during the update process to validate the first commit
+- `hosts` -  host data, specified using the `hosts.json` file. Hosts of the current repo can be specified in its
+parent's repo (meaning that this repo is listed in the parent's `dependencies.json`), or it can be specified in hosts.
+json contained by the repo itself. If hosts data is defined in the parent, it can be propagated to the contained
```

Comment:
```suggestion
parent's repo (meaning that this repo is listed in the parent's `dependencies.json`), or it can be specified in `hosts.json` contained by the repo itself. If hosts data is defined in the parent, it can be propagated to the contained
```